web_poet/page_inputs/client.py

```diff
@@ -45,18 +51,55 @@ class HttpClient:
     def __init__(self, request_downloader: Callable = None):
         self._request_downloader = request_downloader or _perform_request
 
+    @staticmethod
+    def _handle_status(
+        response: HttpResponse,
+        request: HttpRequest,
+        *,
+        allow_status: List[_Status] = None,
+    ) -> None:
+        allow_status_normalized = list(map(str, as_list(allow_status)))
+        allow_all_status = any(
+            [True for s in allow_status_normalized if "*" == s.strip()]
+        )
+
+        if (
+            allow_all_status
+            or response.status is None  # allows serialized responses from tests
+            or response.status < 400
+            or str(response.status) in allow_status_normalized
+        ):
+            return
+
+        status = HTTPStatus(response.status)
+        msg = f"{response.status} {status.name} response for {response.url}"
```

Comment:
just reaslised that this code can be moved to HttpResponseError

Reply:
Good idea. Although I think we could make the default message more generic than this one. There are cases to raise `HttpResponseError` that are not specific to the status code and thus, exceptions messages like this would be invalid. For example, corrupt/malformed/truncated responses even on 200 status.

Updated this on https://github.com/scrapinghub/web-poet/pull/38/commits/43bb5d9998c0e0ca51141f34a5c1b57bf2c6c628.